### PR TITLE
Sanitize the behavior of provided ExceptionMapper classes in dev-mode

### LINKED ISSUE #7883: built-in ExceptionMappers cause unexpected behavior
**Describe the bug**
Quarkus includes several JAX-RS `ExceptionMapper`s out of the box. Those are actually not considered built-in as per the JAX-RS specification, they are just built into Quarkus to improve user experience.

However, they can also make the user experience worse. For example, if I create an `ExceptionMapper` for all `Exception`s, it _isn't_ used if I throw a JAX-RS `NotFoundException`, because Quarkus includes an exception mapper for it (`NotFoundExceptionMapper`). Per the JAX-RS specification, RESTEasy will "use the provider whose generic type is the nearest superclass of the exception."

For example, if my application communicates using JSON and my exception mapper produces some meaningful JSON structure for all `Exception`s, I get an empty response body for `NotFoundException` (which is what `io.quarkus.resteasy.runtime.NotFoundExceptionMapper` does for JSON). It took me a while to figure out what's going on here.

As a workaround, I have to create my own `NotFoundExceptionMapper`.

**Expected behavior**
My own generic exception mapper is used.

**Actual behavior**
Quarkus "built-in" exception mapper is used, because it targets a more specific exception.

I understand this is kinda by design, but unexpected still. I don't know if it would help or not if RESTEasy would consider the "built-in" exception mappers in Quarkus as actually built-in per JAX-RS spec.

**To Reproduce**
1. Add an implementation of `ExceptionMapper<Exception>`.
2. Throw `NotFoundException`.
3. Observe that my own exception mapper is not used.

### PR DESCRIPTION
**Pull request by geoand**
User provided `ExceptionMapper` classes now takes precedence
over the Quarkus provided `NotFoundExceptionMapper`.

This is done in a way that does not change anything but this 
specific behavior, everything else concerning exception handling
continues to work as before.
Note also that this does not change the spec compliance in
any way

Fixes: #7883

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java [hunk 1]
```diff
@@ -94,9 +99,17 @@ public void mapException(Throwable throwable, ResteasyReactiveRequestContext con
             } else {
                 response = exceptionMapper.toResponse(mappedException);
             }
-            context.setResult(response);
-            logBlockingErrorIfRequired(mappedException, context);
-            logNonBlockingErrorIfRequired(mappedException, context);
+            // this special case is used in order to ignore the mapping of built-in mappers and let the exception handling proceed to higher levels
+            if ((response != null) && (IGNORE_RESPONSE.getStatus() == response.getStatus())
```
Comment (Ladicek) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java:103]: Couldn't you just test object identity here? `if (response == IGNORE_RESPONSE) { ... }`

**Comment by Ladicek** [comment]
Nice approach to the fix!

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java (hunk 1 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java:103]: So simple and so true!

🙏

**Review by aldettinger — APPROVED** [review]

**Comment by geoand** [comment]
Definitely something we want for Quarkus 3 @gsmet given how many people have (rightly) complained about this.

**Review by Ladicek — APPROVED** [review]
Not a RESTEasy Reactive expert, but this looks good to me anyway :-)

**Comment by Serkan80** [comment]
This is still not working in Quarkus 3.2.x and I can't override it with the priority attribute in `@ServerExceptionMapper`. And my own exceptionmapper which listens to a `Throwable` also doesn't get triggered on production.

And the name of `NotFoundExceptionMapper` is also misleasding since it listens to all exceptions.

What I expect is that this mapper should be able to be overriden by setting a "higher" priority in a custom mapper.

**Pull request merged**